pkg/controller/daemon/daemoncontroller.go

```diff
@@ -579,23 +735,37 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 			}
 		}
 	}
-	errors := dsc.syncNodes(ds, podsToDelete, nodesNeedingDaemonPods)
+
+	cur, old, err := dsc.constructHistory(ds)
+	if err != nil {
+		return "", fmt.Errorf("failed to construct revisions of DaemonSet: %v", err)
+	}
+
+	err = dsc.cleanupHistory(ds, old, nodeToDaemonPods)
```

Comment:
@kargakis are you suggesting moving `cleanupHistory` to the last step of `syncDaemonSet` (after rolling update)? That makes sense, will do so.